### PR TITLE
[VPP, CM] Corrected RGBP format processing

### PR DESCRIPTION
**Pull request by alariono**
Signed-off-by: Andrey Larionov <andrey.larionov@intel.com>

### CODE REVIEW

File: _studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2018-2019 Intel Corporation
+// Copyright (c) 2018 Intel Corporation
```
Comment (mgonchar) [inline _studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp]: fix year

File: _studio/shared/src/cm_mem_copy.cpp [hunk 2]
```diff
@@ -3221,6 +3225,11 @@ mfxStatus CmCopyWrapper::CopyVideoToSys(mfxFrameSurface1 *pDst, mfxFrameSurface1
 
         mfxI64 verticalPitch = (mfxI64)(pDst->Data.UV - pDst->Data.Y);
```
Comment (mgonchar) [inline _studio/shared/src/cm_mem_copy.cpp]: looks like this code should look in following way:
```
mfxI64 verticalPitch = (mfxI64)(pDst->Data.UV - pDst->Data.Y);
if (pDst->Info.FourCC == MFX_FOURCC_RGBP)
{
    verticalPitch = (mfxI64)(pDst->Data.G - pDst->Data.B);
}
verticalPitch = (verticalPitch % dstPitch)? 0 : verticalPitch / dstPitch;
```

File: _studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp [hunk 3]
```diff
@@ -5214,6 +5214,9 @@ mfxU64 get_background_color(const mfxVideoParam & videoParam)
             return make_def_back_color_yuv(10);
         case MFX_FOURCC_RGB4:
         case MFX_FOURCC_BGR4:
+#if (MFX_VERSION >= 1028)
```
Comment (mgonchar) [inline _studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp]: We have dedicated define ```MFX_ENABLE_RGBP``` I think better to use it

File: _studio/shared/src/cm_mem_copy.cpp (hunk 2 above)
Comment (mgonchar) [reply _studio/shared/src/cm_mem_copy.cpp]: + this code should be covered with define as well:
```
mfxI64 verticalPitch = (mfxI64)(pDst->Data.UV - pDst->Data.Y);
#ifdef MFX_ENABLE_RGBP
if (pDst->Info.FourCC == MFX_FOURCC_RGBP)
{
    verticalPitch = (mfxI64)(pDst->Data.G - pDst->Data.B);
}
#endif
verticalPitch = (verticalPitch % dstPitch)? 0 : verticalPitch / dstPitch;
```

Comment (alariono) [reply _studio/shared/src/cm_mem_copy.cpp]: done

File: _studio/shared/src/cm_mem_copy.cpp [hunk 4]
```diff
@@ -3221,6 +3225,13 @@ mfxStatus CmCopyWrapper::CopyVideoToSys(mfxFrameSurface1 *pDst, mfxFrameSurface1
...
         verticalPitch = (verticalPitch % dstPitch)? 0 : verticalPitch / dstPitch;
+#ifdef MFX_ENABLE_RGBP
+        if (pDst->Info.FourCC == MFX_FOURCC_RGBP)
+        {
+            verticalPitch = (mfxI64)(pDst->Data.G - pDst->Data.B);
+            verticalPitch = (verticalPitch % dstPitch)? 0 : verticalPitch / dstPitch;
```
Comment (mgonchar) [inline _studio/shared/src/cm_mem_copy.cpp]: why do we need two identical lines on 3227 and 3232?

**Pull request merged**